### DISCUSSION TITLE
Check user permission

### DISCUSSION THREAD

**Discussion by mmdbalkhi** [Rapptz/discord.py#7717] [Q&A] [answered]
Hello. How can we check the user's permission and do it if the user has permission to do something?

**Comment by AbstractUmbra** [comment, marked as answer]
There are multiple ways you can do this:

If you want to check their overall permissions within a Guild, you have [`Member.guild_permissions`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.guild_permissions), which is an instance of [`Permissions`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Permissions), so you can check the attributes of it.

If you want their permissions in a specific channel, you have [`GuildChannel.permissions_for`](https://discordpy.readthedocs.io/en/latest/api.html#discord.abc.GuildChannel.permissions_for), which returns the same type.

There is a method that utilises overwrites too if you're interested in that?

**Reply by mmdbalkhi** [reply]
thanks!
